packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1,77 +1,615 @@
+
+          if (!hasSelection) {
+            return "invalid";
+          }
+        }
+
+        return "valid";
+      });
+    }
+  };
+
+  /**
+   * Applies the selected values to the state
+   *
+   * @param {Array} listValues - An array containing the selected values.
+   * @param {Boolean} commitChanges - If `true` the selection if finally committed the dropdown header text should reflect the new selection
+   * @param {Boolean} toggle -If `true` the dropdown should toggle it's current state
+   * @param {Boolean} notifyChanges -If `true` the dropdown will call onChange.
+   */
+  const handleSelection = (
+    listValues,
+    commitChanges,
+    toggle,
+    notifyChanges = true
+  ) => {
+    const selected = getSelected(listValues);
+    if (commitChanges) {
+      setInternalValues(listValues);
+      setSelectionLabel(
+        getSelectionLabel(labels, placeholder, multiSelect, listValues)
+      );
+
+      setValidationState(() => {
+        // this will only run if status is uncontrolled
+        if (required && selected.length === 0) {
+          return "invalid";
+        }
+
+        return "valid";
+      });
+    }
+    if (notifyChanges) onChange?.(multiSelect ? selected : selected[0]);
+    if (toggle) {
+      handleToggle(undefined, false);
+
+      // focus-ring won't be visible even if using the keyboard:
+      // https://github.com/WICG/focus-visible/issues/88
+      dropdownHeaderRef.current?.focus({ preventScroll: true });
+    }
+  };
+
+  /**
+   * Handles the `Cancel` action. Both single and ranged modes are handled here.
+   */
+  const handleCancel = (evt) => {
+    onCancel?.(evt);
+
+    handleToggle(evt, false);
+
+    // focus-ring won't be visible even if using the keyboard:
+    // https://github.com/WICG/focus-visible/issues/88
+    dropdownHeaderRef.current?.focus({ preventScroll: true });
+  };
+
+  const handleClickOutside = (evt) => {
+    onClickOutside?.(evt);
+    onCancel?.(evt);
   };
 
-  const onChangeHandler = (nextValue: string) => {
-    setIsOpen(false);
-    onChange?.(nextValue);
+  const setFocusToContent = (containerRef) => {
+    const inputs = containerRef?.getElementsByTagName("input");
+    if (inputs?.length > 0) {
+      inputs[0].focus();
+      return;
+    }
+    const listItems =
+      containerRef != null ? [...containerRef.getElementsByTagName("li")] : [];
+    listItems.every((listItem) => {
+      if (listItem.tabIndex >= 0) {
+        listItem.focus();
+        return false;
+      }
+      return true;
+    });
   };
 
+  const buildHeaderLabel = () => {
+    const hasSelection = getSelected(internalValues).length > 0;
+    return labels?.select || !multiSelect ? (
+      <StyledTypography
+        as="div"
+        $selectionDisabled={disabled}
+        variant={isOpen || hasSelection ? "normalText" : "placeholderText"}
+        className={clsx(
+          classes?.placeholder,
+          disabled && classes?.selectionDisabled
+        )}
+      >
```

Comment:
We also need to add: `dropdownClasses.placeholder` and `disabled && dropdownClasses.selectionDisabled`.